aiokafka/producer.py

```diff
@@ -496,3 +496,39 @@ def _partition(self, topic, partition, key, value,
         available = list(self._metadata.available_partitions_for_topic(topic))
         return self._partitioner(
             serialized_key, all_partitions, available)
+
+    def create_batch(self, topic, partition=None):
+        """Create and return an empty MessageBatch.
+
+        The batch is not queued for send until submission to ``send_batch``.
+        Use of both ``create_batch`` and ``send`` together is discouraged as it
+        may lead to unexpected results.
+
+        Arguments:
+            topic (str): topic where the batch will be published
```

Comment:
Why do we create the batch for some partition? I would like it to use the default partitioning strategy...

Reply:
Ok, it's optional, saw that... This is a bit of a problem... How above moving those arguments to `send_batch`

Reply:
The batch object requires a TP for instantiation (the TP is later used to populate RecordMetadata objects). I can change that and set the TP during send/drain if you'd prefer.

I'm not sure what you mean by "use the default partitioning strategy" -- batches by their nature are sent to a single partition. The messages within a batch don't get repartitioned during send.

Reply:
It requires TP because you used some internal class for it. If you create a separate one, it will be ok.

Reply:
`use the default partitioning strategy` I wanted to use `partitioner`, but that requires the key of message... Most likely we will require explicit partition to be passed for batches